src/cache.ts

```diff
@@ -44,6 +45,21 @@ export class Cache {
         }
     }
 
+    /**
+     * Given current aspect ratio, determines the max width and height that can
+     * respect the total allowed area for the cache.
+     * @memberOf fabric.util
+     * @param {number} ar aspect ratio
+     * @return {number[]} Limited dimensions X and Y
+     */
+    limitDimsByArea(ar: number) {
+      const { perfLimitSizeTotal } = config;
+      const roughWidth = Math.sqrt(perfLimitSizeTotal * ar);
+      // we are not returning a point on purpose, to avoid circular dependencies
```

Comment:
no is not hacky and not everything should be a point.
Before was x and y but wasn't a point.

Reply:
The reason for the circular dep cycle is just because of the assignment `fabric.Point` that will be removed anyways.
If you assign it in index it will not happen.

Reply:
Don't you agree this is out of pattern?

Reply:
is a function about caching, i don't think returning an array to return 2 values is better or worse than returnin an object and calling it a point. I just really need the 2 values out of there.